clippy_lints/src/utils/mod.rs

```diff
@@ -75,8 +75,8 @@ pub fn parse_msrv(msrv: &str, sess: Option<&Session>, span: Option<Span>) -> Opt
     None
 }
 
-pub fn meets_msrv(msrv: Option<&VersionReq>, lint_msrv: &Version) -> bool {
-    msrv.map_or(true, |msrv| !msrv.matches(lint_msrv))
+pub fn meets_msrv(msrv: Option<&RustcVersion>, lint_msrv: &RustcVersion) -> bool {
+    msrv.map_or(true, |msrv| msrv >= lint_msrv)
```

Comment:
I think we should use the newly implemented `meets` version here. This change should not have an influence on the tests.

```suggestion
    msrv.map_or(true, |msrv| msrv.meets(*lint_msrv))
```

That way our implementation of MSRV is conform with the definition in the [RFC](https://github.com/rust-lang/rfcs/blob/master/text/2523-cfg-path-version.md#cfgversionsemver)

Reply:
Makes sense